alpha/radio.cpp

```diff
@@ -77,29 +114,147 @@ void _TS_RADIO::wifi_enable(bool enable)
 
 void _TS_RADIO::wifi_update()
 {
+  log_d("ESP FREE HEAP: %d", ESP.getFreeHeap());
+    
   if (this->wifiEnabled)
   {
-    if (!wifi_is_connected())
+    if (!this->wifi_is_connected())
     {
       if (millis() - this->wifiTimerStart > WIFI_RETRY_INTERVAL)
       {
         log_i("Connecting to WiFi.");
-        TS_RADIO.wifi_connect();
+        this->wifi_connect();
         this->wifiTimerStart = millis();
       }
       else
       {
         log_i("Waiting for WiFi interval");
       }
+    } else {
+      log_i("Wifi is connected, deinitializing BLE for RAM");
+      log_d("Before ble deinit: %d", ESP.getFreeHeap());
+      TS_HAL.ble_deinit();
+      log_d("After ble deinit: %d", ESP.getFreeHeap());
+      log_i("Downloading tempIds from server");
+      this->download_temp_ids();
     }
   }
   else
   {
     if (wifi_is_connected())
     {
       log_i("Disconnecting WiFi.");
-      TS_RADIO.wifi_disconnect();
+      this->wifi_disconnect();
+      log_i("Initializing BLE.");
+      TS_HAL.ble_init();
     }
   }
   
+}
+
+void _TS_RADIO::config(char* ssid, char* password, char* uid)
+{
+  strcpy(this->ssid, ssid);
+  strcpy(this->password, password);
+  strcpy(this->uid, uid);
+}
+
+void _TS_RADIO::download_temp_ids()
+{
+  log_d("BEGIN LEAKY: %d", ESP.getFreeHeap());
+  WiFiClientSecure *client = new WiFiClientSecure;
+  if (client)
+  {
+    client->setCACert(root_ca);
+    
+    log_d("Connecting to %s", host);
+    if (!client->connect(host, HTTPS_PORT)) 
+    { 
+      log_e("Connection to %s failed", host);
+      return;
+    }
+
+    char body[52];
+    sprintf(body, "{\"data\":{\"uid\":\"%s\"}}", this->uid);
+
+    String postRequest = 
+      String("POST ") + uri_getTempIDs + " HTTP/1.1\r\n" + 
```

Comment:
this part potentially can use a PROGMEM string with placeholders e.g. %s
excessive string concat with + is expensive, supposedly does a string copy of 2 halves to form new string.